### ISSUE TITLE
Experimenting errors in u-boot compilation (I2c_read and I2c_write)

### ISSUE THREAD

**Issue by villalvilla** [mfgbhatti/sun50i-h616-t95-2gb#2]
After applying patch to u-boot and tunning some cross-compile options and installing libraries, I get this error. It seems to be compatibility with the newest PMIC_AXP driver, but not sure about.
Fresh new ideas would help. thanks:

aarch64-linux-gnu-ld.bfd: arch/arm/mach-sunxi/pmic_bus.o: en la función `pmic_bus_read':
/home/projects/v1.0/u-boot/arch/arm/mach-sunxi/pmic_bus.c:86: referencia a `i2c_read' sin definir
aarch64-linux-gnu-ld.bfd: arch/arm/mach-sunxi/pmic_bus.o: en la función `pmic_bus_write':
/home/projects/v1.0/u-boot/arch/arm/mach-sunxi/pmic_bus.c:100: referencia a `i2c_write' sin definir
make[1]: *** [scripts/Makefile.spl:509: spl/u-boot-spl] Error 1
make: *** [Makefile:1982: spl/u-boot-spl] Error 2

**Comment by SavvasMohito** [comment]
Hey, I have the same TV Box 4/64 version with bluetooth and dual band wifi. Did you manage to install the firmware? I can help if you need someone to test this. Let me know!

**Comment by mfgbhatti** [comment]
@SavvasMohito  Hi, There may be a difference in dts for your box.

@villalvilla and @SavvasMohito, I will be very interested in your dts files from the box. your boxes should be rooted i suppose
follow this 
install temux on box
go to /proc/device-tree or /sys/firmware/devicetree/base
install dtc in temux
dtc -I fs -O dts /proc/device-tree # or /sys/firmware/devicetree/base
then pr me with raw dts file.

**Comment by SavvasMohito** [comment]
Hey @mfgbhatti !
The original firmware that came with the box (which is the one I am currently using) is not rooted. I don't have a "root" option in settings like some other boxes. I have tried a custom version of the latest firmware which was rooted but not that stable. I can install it again if root access is required for the dts file you need me to obtain.

**Comment by mfgbhatti** [comment]
yes I think you need root access. Please feel free to share if you find any thing working or think it may work. i am chasing it for some time and learning on my way. thanks @SavvasMohito

**Comment by CSystem77** [comment]
h616 t95 4gb

Model Image : https://ae01.alicdn.com/kf/H70673b97219c4a3fae4fc10dc999ae0f5/Bo-tier-Smart-TV-T95-Android-2-4-HD-6K-Wifi-10-0G-5G-4-go.jpg_640x640.jpg

Link DTS : https://mega.nz/file/8NAwgLxJ#p-t9EEx2L25xqlaS2-mquFvYLb3fk0IaKcGp0vzSr6Y
Link DTB : https://mega.nz/file/pVBFAKZZ#hcIVWSPe_2E7KH-4YrclSQs2ghp_2iHmpUkJ4OiFyWE

Best Regards
CSystem

**Comment by mfgbhatti** [comment]
I will look into it on weekend

**Comment by CSystem77** [comment]
thank you for the help you provide.

**Comment by CSystem77** [comment]
have you any news ? I noticed that my t95 had no bluetooth (Version 4g ram).

**Comment by mfgbhatti** [comment]
hey there, not yet. you got any update??

**Comment by mfgbhatti** [comment]
No activity
